### PR TITLE
Remove eigen submodule

### PR DESCRIPTION
**Pull request by mkristien**
Instead of having eigen library as git submodule, we will store the important headers zipped in a tar file directly in our repo. When the repo is cloned anew, the library is unpacked at the first attempt to `make`. The library installation is done only the first time, unless the tar file is updated.

Updates to the library should be pushed into the repo as a new tar file. The system will recognise the new version and will reinstall the library.

### CODE REVIEW

**Review by brano2 — COMMENTED** [review]
You can probably remove `setup.sh` altogether so it doesn't confuse anyone. If we ever need another submodule, the setup command will be in history and in the old repo.

Otherwise looks good to me. @JustusRudolph maybe you should check you can still use eigen the way you want to? Although I guess you won't really be doing anything much different to what already is in the ported code...

**Comment by mkristien** [comment]
I will be adding `setup.sh` shortly. I want to configure a git hook that will allow commits/push only after a successful build/testing.

File: setup.sh [hunk 1]
```diff
@@ -1 +1,2 @@
-git submodule update --init --recursive --progress
+# We do not longer use submodule for Eigen library
```
Comment (neilweidinger) [inline setup.sh:1]: Can we just delete this file completely?

Comment (brano2) [reply setup.sh:1]: See the discussion above. It will be used again for something else very soon.

**Review by JustusRudolph — APPROVED** [review]
Necessary files are included, no compiler issues, looks fine and seems way more space efficient.

**Pull request merged**